### PR TITLE
Rebase onto Element v1.9.0

### PR DESCRIPTION
**Pull request by giomfo**
fix https://github.com/tchapgouv/tchap-ios/issues/649

### CODE REVIEW

**Comment by giomfo** [comment]
@phlniji I created this PR from your branch in order to start review on it
I did several tests on the application. I noticed only one regression for the moment : the rooms tab is missing (I had only fav and direct tabs at the bottom)
image

File: fastlane/Fastfile [hunk 1]
```diff
@@ -446,19 +393,19 @@ platform :ios do
   end
 
   desc "Upload dsym files to Sentry to symbolicate crashes"
```
Comment (yostyle) [inline fastlane/Fastfile:395]: please comment the description too.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -35,52 +35,6 @@ platform :ios do
...
-    diawi_app_id = URI(diawi_link).path.split('/').last
-    diawi_qr_code_link = "https://www.diawi.com/qrcode/link/#{diawi_app_id}"
-    UI.command_output("Diawi QR code link: " + diawi_qr_code_link.to_s)
-
-    # Set "DIAWI_FILE_LINK" to GitHub environment variables for Github actions
-    if is_ci 
-      sh("echo DIAWI_FILE_LINK=#{diawi_link} >> $GITHUB_ENV")
-      sh("echo DIAWI_QR_CODE_LINK=#{diawi_qr_code_link} >> $GITHUB_ENV")
-    end
-  end
```
Comment (yostyle) [inline fastlane/Fastfile]: please keep this code and comment it if you don't need it

File: Tchap/Modules/Favourites/FavouriteMessages/FavouriteMessagesCoordinator.swift [hunk 3]
```diff
@@ -49,7 +49,7 @@ final class FavouriteMessagesCoordinator: FavouriteMessagesCoordinatorType {
         formatter.treatMatrixUserIdAsLink = true
         formatter.treatMatrixRoomIdAsLink = true
         formatter.treatMatrixRoomAliasAsLink = true
-        formatter.treatMatrixGroupIdAsLink = true
+//        formatter.treatMatrixGroupIdAsLink = true
```
Comment (yostyle) [inline Tchap/Modules/Favourites/FavouriteMessages/FavouriteMessagesCoordinator.swift:52]: What is the default value ? It's not possible to set this variable ?

File: Tchap/Generated/Storyboards_Riot.swift [hunk 4]
```diff
@@ -147,6 +147,11 @@ internal extension StoryboardScene {
 
     internal static let initialScene = InitialSceneType<Tchap.ReactionHistoryViewController>(storyboard: ReactionHistoryViewController.self)
   }
+  enum RoomContextPreviewViewController: StoryboardType {
```
Comment (yostyle) [inline Tchap/Generated/Storyboards_Riot.swift:150]: Could we ignore this file in git repo ?

File: Tchap/Generated/Strings.swift [hunk 5]
```diff
@@ -3,456 +3,456 @@
 
 import Foundation
 
-// swiftlint:disable superfluous_disable_command file_length implicit_return
+// swiftlint:disable superfluous_disable_command file_length implicit_return prefer_self_in_static_references
```
Comment (yostyle) [inline Tchap/Generated/Strings.swift:6]: Could we ignore this file in git repo ?

File: Tchap/Generated/Storyboards.swift [hunk 6]
```diff
@@ -10,7 +10,8 @@ import UIKit
 
 // MARK: - Storyboard Scenes
 
-// swiftlint:disable explicit_type_interface identifier_name line_length type_body_length type_name
+// swiftlint:disable explicit_type_interface identifier_name line_length prefer_self_in_static_references
```
Comment (yostyle) [inline Tchap/Generated/Storyboards.swift:13]: Could we ignore this file in git repo ?

File: Tchap/Generated/InfoPlist.swift [hunk 7]
```diff
@@ -11,7 +11,6 @@ import Foundation
 // swiftlint:disable identifier_name line_length type_body_length
 internal enum InfoPlist {
   private static let _document = PlistDocument(path: "Info.plist")
-
```
Comment (yostyle) [inline Tchap/Generated/InfoPlist.swift:14]: Could we ignore this file in git repo ?

File: Tchap/Config/BuildSettings.swift [hunk 8]
```diff
@@ -252,7 +252,7 @@ final class BuildSettings: NSObject {
     static let allowInviteExernalUsers: Bool = true
     
     // MARK: - Side Menu
-    static let enableSideMenu: Bool = true
+    static let enableSideMenu: Bool = true && !newAppLayoutEnabled
```
Comment (yostyle) [inline Tchap/Config/BuildSettings.swift:255]: Should we have this update in element ios ?

File: Riot/Modules/TabBar/TabBarCoordinator.swift [hunk 9]
```diff
@@ -1034,13 +925,13 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
             return
         }
 
-        masterTabBarController.homeViewController.startActivityIndicator()
+//        masterTabBarController.homeViewController.startActivityIndicator()
```
Comment (yostyle) [inline Riot/Modules/TabBar/TabBarCoordinator.swift:928]: Why did you remove the indicator ?

**Comment by yostyle** [comment]
> @phlniji I created this PR from your branch in order to start review on it I did several tests on the application. I noticed only one regression for the moment : the rooms tab is missing (I had only fav and direct tabs at the bottom) image

Could you restart the application (kill it and start it) ?

File: Config/BuildSettings.swift [hunk 10]
```diff
@@ -390,8 +390,8 @@ final class BuildSettings: NSObject {
     static let authEnableRefreshTokens = false
     
     // MARK: - Onboarding
-    static let onboardingShowAccountPersonalization = true
-    static let onboardingEnableNewAuthenticationFlow = true
+    static let onboardingShowAccountPersonalization = false
+    static let onboardingEnableNewAuthenticationFlow = false
```
[the remark this answers is not in the document]
Comment (giomfo) [reply Config/BuildSettings.swift:394]: this is something that we can consider if this eases the integration of the new password recovery flow, and if there is a way to customize it by including the Tchap login/register flow
but this is not a priority

File: Riot/Modules/Common/Recents/RecentsViewController.m [hunk 11]
```diff
@@ -2695,11 +2684,6 @@ - (void)toggleSearchBar {
             [self cleanSearchAndHideSearchBar:TRUE];
         }
     }
-<<<<<<< HEAD
-=======
-    
-    return [self.contextMenuProvider contextMenuConfigurationWith:cellData from:cell session:self.dataSource.mxSession];
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Common/Recents/RecentsViewController.m]: It's an auto-merge issue.
This is why the code is deleted here (but in fact still available at line 2720).

File: Riot/Modules/TabBar/TabBarCoordinator.swift (hunk 9 above)
Comment (phlniji) [reply Riot/Modules/TabBar/TabBarCoordinator.swift:928]: There is no homeViewController in Tchap. So I forgot to add the tchap comment. I will add it.

File: Tchap/Config/BuildSettings.swift (hunk 8 above)
Comment (phlniji) [reply Tchap/Config/BuildSettings.swift:255]: This code is from Element iOS.

**Comment by phlniji** [comment]
> > @phlniji I created this PR from your branch in order to start review on it I did several tests on the application. I noticed only one regression for the moment : the rooms tab is missing (I had only fav and direct tabs at the bottom) image
> 
> Could you restart the application (kill it and start it) ? -> @yostyle FYI I did, the problem is persistent

A part of the code for the room tab has been commented by error. It will be fixed in next commit.

File: Riot/Modules/Common/Recents/Service/MatrixSDK/RecentsListService.swift [hunk 12]
```diff
@@ -467,16 +458,11 @@ public class RecentsListService: NSObject, RecentsListServiceProtocol {
     private var shouldShowFavorited: Bool {
         return fetcherTypesForMode[mode]?.contains(.favorited) ?? false
     }
-
+    
     private var shouldShowDirect: Bool {
         switch mode {
-<<<<<<< HEAD
-//        case .home:
+//        case .home, .allChats:
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Common/Recents/Service/MatrixSDK/RecentsListService.swift:464]: Will be fixed in next commit

File: Riot/Modules/Room/VoiceMessages/VoiceMessageAudioConverter.swift [hunk 13]
```diff
@@ -15,7 +15,7 @@
 //
 
 import Foundation
-import SwiftOGG
+//import SwiftOGG
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Room/VoiceMessages/VoiceMessageAudioConverter.swift:18]: Will be fixed in next commit

File: Riot/Modules/Common/Recents/Service/MatrixSDK/RecentsListService.swift [hunk 14]
```diff
@@ -185,23 +176,23 @@ public class RecentsListService: NSObject, RecentsListServiceProtocol {
...
-                                                 favoriteTag: false,
-                                                 suggested: false,
-                                                 alphabetical: true,
-                                                 missedNotificationsFirst: false,
-                                                 unreadMessagesFirst: false)
-            case .activity:
-                return MXRoomListDataSortOptions(missedNotificationsFirst: pinMissed,
-                                                 unreadMessagesFirst: pinUnread)
-            @unknown default:
+//            switch AllChatsLayoutSettingsManager.shared.allChatLayoutSettings.sorting {
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Common/Recents/Service/MatrixSDK/RecentsListService.swift:179]: Will be fixed in next commit

File: Tchap/Generated/Strings.swift (hunk 5 above)
Comment (phlniji) [reply Tchap/Generated/Strings.swift:6]: It is a good question.
For the moment, I think it is better to have the same behavior as Element, but we may discuss about this in a near future.

File: Tchap/Generated/Storyboards.swift (hunk 6 above)
Comment (phlniji) [reply Tchap/Generated/Storyboards.swift:13]: Same as previous comment

File: Tchap/Generated/InfoPlist.swift (hunk 7 above)
Comment (phlniji) [reply Tchap/Generated/InfoPlist.swift:14]: Same as previous comment

**Review by giomfo — COMMENTED** [review]
LGTM - approved

**Review by yostyle — APPROVED** [review]
